example/centos/centos-9-x86_64-qcow2.yaml

```diff
@@ -251,130 +243,39 @@ otk.target.osbuild:
-                start: 413696
-                size: 2097152
-                lock: true
-        - type: org.osbuild.mkfs.xfs
-          options:
-            uuid: dde1466f-f75b-4f05-bcf6-8386564c6f79
-            label: root
-          devices:
-            device:
-              type: org.osbuild.loopback
-              options:
-                filename: disk.img
-                start: 2510848
-                size: 18460639
-                lock: true
-        - type: org.osbuild.copy
-          inputs:
-            root-tree:
-              type: org.osbuild.tree
-              origin: org.osbuild.pipeline
-              references:
-                - name:os
-          options:
-            paths:
-              - from: input://root-tree/
-                to: mount://-/
-          devices:
-            '-':
-              type: org.osbuild.loopback
-              options:
-                filename: disk.img
-                start: 2510848
-                size: 18460639
-            boot:
-              type: org.osbuild.loopback
-              options:
-                filename: disk.img
-                start: 413696
-                size: 2097152
-            boot-efi:
-              type: org.osbuild.loopback
-              options:
-                filename: disk.img
-                start: 4096
-                size: 409600
-          mounts:
-            - name: '-'
-              type: org.osbuild.xfs
-              source: '-'
-              target: /
-            - name: boot
-              type: org.osbuild.xfs
-              source: boot
-              target: /boot
-            - name: boot-efi
-              type: org.osbuild.fat
-              source: boot-efi
-              target: /boot/efi
-        - type: org.osbuild.grub2.inst
-          options:
-            filename: disk.img
-            platform: i386-pc
-            location: 2048
-            core:
-              type: mkimage
-              partlabel: gpt
-              filesystem: xfs
-            prefix:
-              type: partition
-              partlabel: gpt
-              number: 2
-              path: /grub2
+        otk.op.join:
+          values:
+            - otk.external.otk-make-partition-stages:
+                ${filesystem}
+            - - type: org.osbuild.copy
+                inputs:
+                  root-tree:
+                    type: org.osbuild.tree
+                    origin: org.osbuild.pipeline
+                    references:
+                      - name:os
+                options:
+                  paths:
+                    - from: input://root-tree/
+                      to: mount://-/
+                devices:
+                  ${fs_options.devices}
+                mounts:
+                  ${fs_options.mounts}
+              - type: org.osbuild.grub2.inst
+                options:
+                  filename: disk.img
+                  platform: i386-pc
+                  location: 2048
```

Comment:
These will also have to be somehow defined via otk-gen-partition I think

Reply:
We can probably look into the usability/niceness of it better after we have some working examples so I wouldn't block this PR on this :)

Reply:
Agreed, I opened https://github.com/osbuild/images/pull/916 but the change can be a folloup.